src/platforms/hosted/serial_unix.c

```diff
@@ -43,6 +47,73 @@ static uint8_t read_buffer[READ_BUFFER_LENGTH];
 static size_t read_buffer_fullness = 0U;
 static size_t read_buffer_offset = 0U;
 
+/* Socket code taken from https://beej.us/guide/bgnet/ */
+static bool try_opening_network_device(const char *const name)
+{
+	// Maximum legal length of a hostname
+	char hostname[256];
+
+	// The service name or port number
+	char *service_name;
+
+	struct addrinfo addr_hints = {
+		.ai_family = AF_UNSPEC,
+		.ai_socktype = SOCK_STREAM,
+		.ai_protocol = IPPROTO_TCP,
+		.ai_flags = AI_ADDRCONFIG | AI_V4MAPPED,
+	};
+	struct addrinfo *server_info;
+
+	if (!name) {
+		return false;
+	}
+
+	// Copy the hostname to an internal array. We need to modify it
+	// to separate the hostname from the service name.
+	strncpy(hostname, name, sizeof(hostname) - 1U);
+
+	service_name = strstr(hostname, ":");
+	if (service_name == NULL) {
+		return false;
```

Comment:
It might be helpful to `DEBUG_WARN` why this code is ignoring the name string as not-a-`hostname:port`.

Reply:
That will get printed out as part of every invocation. Is that okay? I thought this might be a niche case so I wanted to avoid reporting anything like that.

Reply:
Given that should only happen if BMDA was unable to open the target as a file/device and it's then failing as a network socket target - that seems completely reasonable as it helps the user diagnose what's going on.